<launch>
  <!-- Frames of reference. -->
  <arg name="fixed_frame" default="world" />
  <arg name="sensor0_frame" default="robot0_laser_0" />
  <arg name="sensor1_frame" default="robot1_laser_0" />
  <arg name="sensor2_frame" default="robot2_laser_0" />
  <arg name="sensor3_frame" default="robot3_laser_0" />
  <arg name="sensor4_frame" default="robot4_laser_0" />
  <arg name="sensor5_frame" default="robot5_laser_0" />
  <arg name="sensor6_frame" default="robot6_laser_0" />
  <arg name="sensor7_frame" default="robot7_laser_0" />

  <!-- Topics. -->
  <arg name="sensor0_topic" default="/robot0/laser_0" />
  <arg name="sensor1_topic" default="/robot1/laser_0" />
  <arg name="sensor2_topic" default="/robot2/laser_0" />
  <arg name="sensor3_topic" default="/robot3/laser_0" />
  <arg name="sensor4_topic" default="/robot4/laser_0" />
  <arg name="sensor5_topic" default="/robot5/laser_0" />
  <arg name="sensor6_topic" default="/robot6/laser_0" />
  <arg name="sensor7_topic" default="/robot7/laser_0" />
  <arg name="vis_topic" default="/vis/map" />

  <!-- Dimensions and bounds. -->
  <arg name="x_num" default="96" />
  <arg name="x_min" default="0" />
  <arg name="x_max" default="24" />
  <arg name="y_num" default="56" />
  <arg name="y_min" default="0" />
  <arg name="y_max" default="14" />

  <!-- Update parameters. -->
  <arg name="occupied_update" default="0.7" />
  <arg name="occupied_threshold" default="0.97" />
  <arg name="free_update" default="0.3" />
  <arg name="free_threshold" default="0.03" />

  <!-- Random fraction of scan rays to process (a hack for speed). -->
  <arg name="random_downsample" default="0.1" />

  <!-- Mapper node. -->
  <node name="mapper"
        pkg="among_us"
        type="mapping_node.py"
        output="screen">

    <param name="frames/fixed" value="$(arg fixed_frame)" />
    <param name="frames/sensor0" value="$(arg sensor0_frame)" />
    <param name="frames/sensor1" value="$(arg sensor1_frame)" />
    <param name="frames/sensor2" value="$(arg sensor2_frame)" />
    <param name="frames/sensor3" value="$(arg sensor3_frame)" />
    <param name="frames/sensor4" value="$(arg sensor4_frame)" />
    <param name="frames/sensor5" value="$(arg sensor5_frame)" />
    <param name="frames/sensor6" value="$(arg sensor6_frame)" />
    <param name="frames/sensor7" value="$(arg sensor7_frame)" />

    <param name="topics/vis" value="$(arg vis_topic)" />
    <param name="topics/sensor0" value="$(arg sensor0_topic)" />
    <param name="topics/sensor1" value="$(arg sensor1_topic)" />
    <param name="topics/sensor2" value="$(arg sensor2_topic)" />
    <param name="topics/sensor3" value="$(arg sensor3_topic)" />
    <param name="topics/sensor4" value="$(arg sensor4_topic)" />
    <param name="topics/sensor5" value="$(arg sensor5_topic)" />
    <param name="topics/sensor6" value="$(arg sensor6_topic)" />
    <param name="topics/sensor7" value="$(arg sensor7_topic)" />

    <param name="x/num" value="$(arg x_num)" />
    <param name="x/min" value="$(arg x_min)" />
    <param name="x/max" value="$(arg x_max)" />
    <param name="y/num" value="$(arg y_num)" />
    <param name="y/min" value="$(arg y_min)" />
    <param name="y/max" value="$(arg y_max)" />

    <param name="update/occupied" value="$(arg occupied_update)" />
    <param name="update/occupied_threshold" value="$(arg occupied_threshold)" />
    <param name="update/free" value="$(arg free_update)" />
    <param name="update/free_threshold" value="$(arg free_threshold)" />

    <param name="random_downsample" value="$(arg random_downsample)" />
  </node>
</launch>